projects/packages/backup/src/class-rest-controller.php

```diff
@@ -138,4 +255,277 @@ public static function delete_backup_helper_script( $request ) {
 			)
 		);
 	}
+
+	/**
+	 * Fetch a backup of a database object, along with all of its metadata.
+	 *
+	 * @access public
+	 * @static
+	 *
+	 * @param WP_REST_Request $request The request sent to the WP REST API.
+	 * @return array
+	 */
+	public static function fetch_database_object_backup( $request ) {
+		global $wpdb;
+
+		// Disable Sync as this is a read-only operation and triggered by sync activity.
+		Sync_Actions::mark_sync_read_only();
+
+		$allowed_object_types = self::get_allowed_object_types();
+		// Safe to do this as we have already validated the object_type key exists in self::get_allowed_object_types().
+		$object_type = $allowed_object_types[ $request->get_param( 'object_type' ) ];
+		$object_id   = $request->get_param( 'object_id' );
+		$table       = $wpdb->prefix . $object_type['table'];
+		$id_field    = $object_type['id_field'];
+
+		// Fetch the requested object.
+		$object = $wpdb->get_row(
+			$wpdb->prepare(
+				'SELECT * FROM `' . $table . '` WHERE `' . $id_field . '` = %d', // phpcs:ignore WordPress.DB.PreparedSQL.NotPrepared
+				$object_id
+			)
+		);
+
+		if ( empty( $object ) ) {
+			return new WP_Error( 'object_not_found', __( 'Object not found', 'jetpack' ), array( 'status' => 404 ) );
+		}
+
+		$result = array( 'object' => $object );
+
+		// Fetch associated metadata (if this object type has any).
+		if ( ! empty( $object_type['meta_type'] ) ) {
+			$result['meta'] = get_metadata( $object_type['meta_type'], $object_id );
+		}
+
+		// If there is a child linked table (eg: woocommerce_tax_rate_locations), fetch linked records.
+		if ( ! empty( $object_type['child_table'] ) ) {
+			$child_table    = $wpdb->prefix . $object_type['child_table'];
+			$child_id_field = $object_type['child_id_field'];
+
+			$result['children'] = $wpdb->get_results(
+				$wpdb->prepare(
+					'SELECT * FROM `' . $child_table . '` where `' . $child_id_field . '` = %d', // phpcs:ignore WordPress.DB.PreparedSQL.NotPrepared
```

Comment:
I'm not sure I understand why the phpcs:ignore was needed here and above?

Reply:
Good question, Jeremy, that was my initial approach as well.
Essentially I took this (existing code in `class.jetpack-json-api-get-database-object-backup-endpoint.php`):

```
// Fetch the requested object
$query  = $wpdb->prepare( 'select * from `' . $table . '` where `' . $id_field . '` = %d', $this->object_id );
$object = $wpdb->get_row( $query );
```

and turned it into this:
```
$object = $wpdb->get_row(
	$wpdb->prepare(
			'select * from %s where %s = %d',
			$table,
			$id_field,
			$object_id
		)
	);
)
```

but while testing on a JN site, I realized that this produced a db table not-exists error, produced by the single quotes appended before and after the table name in the resulting query.
Given we already validate the `table` input and only allow a specific enum of table names and then use internal logic to grab the `$id_field` as well, I believe we are on the clear in terms of security.

Let me know if there's a better approach!